pkg/xray/resource_xray_watch_test.go

```diff
@@ -219,6 +219,69 @@ func TestAccWatch_multipleBuilds(t *testing.T) {
 	})
 }
 
+func TestAccWatch_allProjects(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	testData := make(map[string]string)
+	copyStringMap(testDataWatch, testData)
+
+	testData["resource_name"] = resourceName
+	testData["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	testData["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	testData["watch_type"] = "all-projects"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		CheckDestroy:      verifyDeleted(fqrn, testCheckWatch),
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, allProjectsWatchTemplate, testData),
+				Check:  verifyXrayWatch(fqrn, testData),
+			},
+		},
+	})
+}
+
+func TestAccWatch_singleProject(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	testData := make(map[string]string)
+	copyStringMap(testDataWatch, testData)
+
+	testData["resource_name"] = resourceName
+	testData["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	testData["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	testData["watch_type"] = "project"
+	testData["project_name_0"] = "test-project11"
+	testData["project_name_1"] = "test-project1111"
+	testData["project_key_0"] = "test11"
+	testData["project_key_1"] = "test111"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccCreateProject(t, testData["project_key_0"], testData["project_name_0"])
+			testAccCreateProject(t, testData["project_key_1"], testData["project_name_1"])
+		},
+		CheckDestroy: verifyDeleted(fqrn, func(id string, request *resty.Request) (*resty.Response, error) {
+			testAccDeleteProject(t, testData["project_key_0"])
```

Comment:
Any one of these could produce an error. Because it's a test, it probably is ok. But you should be checking every call

Reply:
testAccDeleteProject will show the error. Do you want these functions to return `(*resty.Response, error)`? 
Not sure if it's necessary for these calls though.

Reply:
Anyone one of them could produce and error. And, even if you didn't check the error state, likely something later would fail. 

I know it's tedious, but that's golang for you. I'll let @alexhung give his .02 USD

Reply:
Please check the current implementation. Positive and negative tests passed.